pkg/zfs/volume.go

```diff
@@ -290,10 +286,6 @@ func UpdateSnapInfo(snap *apis.ZFSSnapshot) error {
 	finalizers := []string{ZFSFinalizer}
 	labels := map[string]string{ZFSNodeKey: NodeID}
 
-	if snap.Finalizers != nil {
```

Comment:
Why was this removed? If user has added a custom finalizer and snapshot is not ready, will this mark the snapshot CR as ready?

Reply:
if this check is there, snapshot will never be ready, so we have to remove as it was assumed that no one will create the resources manually.